FirebaseMessaging/Sources/Token/FIRMessagingTokenManager.m

```diff
@@ -458,41 +457,40 @@ - (void)deleteWithHandler:(void (^)(NSError *))handler {
  *  since the Keychain items are marked with `*BackupThisDeviceOnly`.
  */
 - (void)resetCredentialsIfNeeded {
-  BOOL checkinPlistExists = [_checkinStore hasCheckinPlist];
+  BOOL checkinPlistExists = [_authService hasCheckinPlist];
   // Checkin info existed in backup excluded plist. Should not be a fresh install.
   if (checkinPlistExists) {
     return;
   }
-
-  // Resets checkin in keychain if a fresh install.
   // Keychain can still exist even if app is uninstalled.
-  FIRMessagingCheckinPreferences *oldCheckinPreferences = [_checkinStore cachedCheckinPreferences];
-
-  if (oldCheckinPreferences) {
-    [_checkinStore removeCheckinPreferencesWithHandler:^(NSError *error) {
```

Comment:
Note for reviewers: The _authService wraps _checkinService and the replacement call `[_authService resetCheckinWithHandler]` internally calls this method `[_checkinStore removeCheckinPreferencesWithHandler]` internally.